src/waveform/waveformwidgetfactory.cpp

```diff
@@ -432,6 +441,22 @@ void WaveformWidgetFactory::setZoomSync(bool sync) {
     }
 }
 
+void WaveformWidgetFactory::setDisplayBeatGrid(bool sync) {
+    m_beatGridEnabled = sync;
+    if (m_config) {
+        m_config->set(ConfigKey("[Waveform]", "beatGridLinesCheckBox"), ConfigValue(m_beatGridEnabled));
```

Comment:
Oh, I see the way you did it is consistent with the surrounding code. You don't have to clean this up if you don't have time, but it would be nice to move the setting of the ConfigKeys into DlgPrefWaveform.